examples/DisposableTokens/Program.cs

```diff
@@ -5,38 +5,85 @@
-    ExpiresIn.Minutes(5)
-);
-
-if (tokenResponse is GenerateDisposableTokenResponse.Success token)
-{
-    Console.WriteLine("The generated disposable token is: " + token.AuthToken);
-    Console.WriteLine("The token endpoint is: " + token.Endpoint);
-    Console.WriteLine("The token expires at (epoch timestamp): " + token.ExpiresAt.Epoch());
-}
-else if (tokenResponse is GenerateDisposableTokenResponse.Error err)
-{
-    Console.WriteLine("Error generating disposable token: " + err.Message);
+
+    private static readonly ICredentialProvider authProvider;
+    private static readonly IAuthClient client;
+ 
+    static Program()
+    {
+        authProvider = new EnvMomentoTokenProvider("MOMENTO_AUTH_TOKEN");
+        client = new AuthClient(AuthConfigurations.Default.Latest(), authProvider);
+    }
+
+    private static async Task<GenerateDisposableTokenResponse.Success> GenerateTokenWithEnumeratedPermissions()
+    {
+        var scope = new DisposableTokenScope(Permissions: new List<DisposableTokenPermission>
+        {
+            new DisposableToken.CacheItemPermission(
+                CacheRole.ReadWrite,
+                CacheSelector.ByName("cache"),
+                CacheItemSelector.AllCacheItems
+            ),
+            new DisposableToken.CachePermission(
+                CacheRole.ReadOnly,
+                CacheSelector.ByName("topsecret")
+            ),
+            new DisposableToken.TopicPermission(
+                TopicRole.PublishSubscribe,
+                CacheSelector.ByName("cache"),
+                TopicSelector.ByName("topic")
+            )
+        });
+
+        var tokenResponse = await client.GenerateDisposableTokenAsync(
+            scope,
+            ExpiresIn.Minutes(30)
+        );
+        return ProcessResponse(tokenResponse);
+    }
+
+    private static async Task<GenerateDisposableTokenResponse.Success> GenerateTokenWithPredefinedScope()
+    {
+        var tokenResponse = await client.GenerateDisposableTokenAsync(
+            // many predefined scopes are available to easily control access to caches, cache keys, and topics
+            DisposableTokenScopes.CacheKeyReadWrite("cache", "cache-key"),
+            ExpiresIn.Minutes(30)
+        );
+        return ProcessResponse(tokenResponse);
+    }
+
+    private static GenerateDisposableTokenResponse.Success ProcessResponse(GenerateDisposableTokenResponse tokenResponse)
+    {
+        if (tokenResponse is GenerateDisposableTokenResponse.Success token)
+        {
+            return token;
+        }
+        if (tokenResponse is GenerateDisposableTokenResponse.Error err)
+        {
+            throw err.InnerException;
+        }
+        throw new Exception("Unknown response type: " + tokenResponse);
+    }
+
+    private static void PrintTokenInfo(GenerateDisposableTokenResponse.Success token)
+    {
+        Console.WriteLine(
+            "The generated disposable token (truncated): " 
+            + token.AuthToken.Substring(0, 10) 
```

Comment:
It's far too late for .NET, but for the next sdk we make, we should make a token type that overrides toString() so that it only prints in a truncated way.

Reply:
I'm adding a `ToString()` override to `GenerateDisposableTokenResponse.Success` in a separate PR right now!